hikari/files.py

```diff
@@ -470,6 +480,40 @@ async def read(
         async with self.stream(executor=executor) as reader:
             return await reader.read()
 
+    async def save(
+        self,
+        path: typing.Optional[Pathish] = None,
+        *,
+        executor: typing.Optional[concurrent.futures.Executor] = None,
+        force: bool = False,
+    ) -> None:
+        """Save this resource to disk.
+
+        This writes the resource file in chunks, and so does not load
+        the entire resource into memory.
+
+        Parameters
+        ----------
+        path : typing.Optional[Pathish]
+            The path to save this resource to. If this is a string, the
+            path will be relative to the current working directory. If
+            `None`, the resource will be saved as `filename` in
+            the current working directory.
+        executor : typing.Optional[concurrent.futures.Executor]
+            The executor to run in for blocking operations.
+            If `None`, then the default executor is used for
+            the current event loop.
+        force : bool
+            Whether to overwrite an existing file. Defaults to `False`.
+        """
+        loop = asyncio.get_running_loop()
+        path = ensure_path(path or self.filename)
+
+        with await loop.run_in_executor(executor, _open_write_path, path, self.filename, force) as f:
```

Comment:
Closing a file can also be a blocking operation so you might want to do something like this instead:
```py
file = await loop.run_in_executor(executor, _open_write_path, path, self.filename, force)

try:
    ...

finally:
    await loop.run_in_executor(file.close)
```

Reply:
Forgot about that small detail, thanks for pointing it out!

Reply:
Changed!